pkg/kubelet/cm/node_container_manager_linux_test.go

```diff
@@ -258,6 +258,75 @@ func TestNodeAllocatableForEnforcement(t *testing.T) {
 	}
 }
 
+func TestVerifyReservedResourcesHaveCapacity(t *testing.T) {
+	testCases := []struct {
+		kubeReserved         v1.ResourceList
+		systemReserved       v1.ResourceList
+		capacity             v1.ResourceList
+		invalidConfiguration bool
+		name                 string
+	}{
+		{
+			name:           "supported reserved 2Mi huge pages resources",
+			kubeReserved:   getHugepagesResourceList("2Mi", "2Mi"),
+			systemReserved: getHugepagesResourceList("2Mi", "2Mi"),
+			capacity:       getHugepagesResourceList("2Mi", "1Gi"),
+		},
+		{
+			name:           "supported reserved cpu and memory resources",
+			kubeReserved:   getResourceList("1", "200Mi"),
+			systemReserved: getResourceList("1", "500Mi"),
+			capacity:       getResourceList("24", "76Gi"),
+		},
+		{
+			name:           "no reserved resources",
+			kubeReserved:   v1.ResourceList{},
+			systemReserved: v1.ResourceList{},
+			capacity:       getResourceList("10", "10Gi"),
+		},
+		{
+			name:           "supported 64Ki huge pages resources",
+			kubeReserved:   getHugepagesResourceList("64Ki", "2Mi"),
+			systemReserved: getHugepagesResourceList("64Ki", "2Mi"),
+			capacity:       getHugepagesResourceList("64Ki", "1Gi"),
+		},
+		{
+			name:                 "unsupported reserved 1Gi huge pages resources",
+			kubeReserved:         getHugepagesResourceList("1Gi", "2Gi"),
+			systemReserved:       getHugepagesResourceList("2Mi", "1Mi"),
+			capacity:             getHugepagesResourceList("2Mi", "1Gi"),
+			invalidConfiguration: true,
+		},
+		{
+			name:                 "unsupported reserved 1Gi and 2Mi huge pages resources",
+			kubeReserved:         getHugepagesResourceList("1Gi", "2Gi"),
+			systemReserved:       getHugepagesResourceList("2Mi", "1Mi"),
+			capacity:             getResourceList("6", "32Gi"),
+			invalidConfiguration: true,
+		},
```

Comment:
how about a test case like this, where it should fail?
    kubeReserved:         getHugepagesResourceList("1Gi", "1Gi"),
    systemReserved:       getHugepagesResourceList("1Gi", "1Gi"),
    capacity:             getResourceList("1Gi", "1Gi"),

Reply:
```
kubeReserved: getHugepagesResourceList("1Gi", "1Gi"),
systemReserved: getHugepagesResourceList("1Gi", "1Gi"),
capacity: getResourceList("1Gi", "1Gi"),
```

Will fail since the the node capacity doesn't support `hugepages-1Gi`. (and `1Gi` of cpu doesn't make that much sense).

If you are thinking about the situation where more memory is reserved than actually is available on the node, that would be caught elsewhere (`cm.validateNodeAllocatable`).

Reply:
But thanks for mentioning it. Added a test case here: https://github.com/kubernetes/kubernetes/pull/83541/files#diff-1095a9691506e55223999753e484c972R485-R488